cpu/sam0_common/periph/pwm.c

```diff
@@ -126,11 +73,33 @@ static uint8_t get_prescaler(unsigned int target, int *scale)
 
 static void poweron(pwm_t dev)
 {
-    PM->APBCMASK.reg |= _apbcmask_tcc(dev);
-    GCLK->CLKCTRL.reg = (GCLK_CLKCTRL_CLKEN |
-                         GCLK_CLKCTRL_GEN_GCLK0 |
-                         GCLK_CLKCTRL_ID(_clk_id(dev)));
-    while (GCLK->STATUS.bit.SYNCBUSY) {}
+    const pwm_conf_t *cfg = &pwm_config[dev];
+
+    sam0_gclk_enable(cfg->gclk_src);
+#ifdef MCLK
+    GCLK->PCHCTRL[cfg->tim.gclk_id].reg = GCLK_PCHCTRL_GEN(cfg->gclk_src)
+                                        | GCLK_PCHCTRL_CHEN;
+    *cfg->tim.mclk |= cfg->tim.mclk_mask;
+#else
+    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_CLKEN
+                      | GCLK_CLKCTRL_GEN(cfg->gclk_src)
+                      | GCLK_CLKCTRL_ID(cfg->tim.gclk_id);
+    PM->APBCMASK.reg |= cfg->tim.pm_mask;
+#endif
+}
+
+static void poweroff(pwm_t dev)
+{
+    const pwm_conf_t *cfg = &pwm_config[dev];
+
+#ifdef MCLK
+    GCLK->PCHCTRL[cfg->tim.gclk_id].reg = 0;
+    *cfg->tim.mclk &= ~cfg->tim.mclk_mask;
+#else
+    PM->APBCMASK.reg &= ~cfg->tim.pm_mask;
+    GCLK->CLKCTRL.reg = GCLK_CLKCTRL_GEN_GCLK7
```

Comment:
Why GEN_GCLK7 ?

Reply:
I guess the idea was to connect it to an unused GCLK so the original one can be turned off if it runs ondemand.
This was there already, I just kept that part - I can check if it makes any difference to power consumption (if something other than `SAM0_GCLK_MAIN` is used), I'd be happy to drop it.